selene_sdk/samplers/multi_file_sampler.py

```diff
@@ -3,45 +3,56 @@
 FileSampler for each mode of sampling (train, test, validation).
 The MultiFileSampler is therefore a subclass of Sampler.
 """
+import numpy as np
+from torch.utils.data import DataLoader
 
 from .sampler import Sampler
 
 
 class MultiFileSampler(Sampler):
     """
-    This sampler contains individual file samplers for each mode.
-    The file samplers parse .bed/.mat files that correspond to
-    training, validation, and testing and MultiFileSampler calls on
-    the correct file sampler to draw samples for a given mode.
-
-    Attributes
+    This sampler draws samples from individual file samplers or data loaders
+    that corresponds to training, validation, and testing (optional) modes.
+    MultiFileSampler calls on the correct file sampler or data loader to draw
+    samples for a given mode. Example file samples are under
+    `selene_sdk.samplers.file_samplers` and example data loaders are under
+    `selene_sdk.samplers.dataloaders`.
+
+    MultiFileSampler can use either file samplers or data loaders for
+    different modes. Using data loaders for some modes while using file samplers
+    for other modes are also allowed. The file samplers parse data files
+    (e.g. bed, mat, or hdf5). The data loaders provide multi-worker iterators
+    that draw samples from a sampler or a file sampler. As data loaders
+    support parallel sampling, they are generally recommended for sampling speed.
+
+    Parameters
     ----------
-    train_sampler : selene_sdk.samplers.file_samplers.FileSampler
-        Load your training data as a `FileSampler` before passing it
-        into the `MultiFileSampler` constructor.
-    validate_sampler : selene_sdk.samplers.file_samplers.FileSampler
-        The validation dataset file sampler.
-    features : list(str)
-        The list of features the model should predict
-    test_sampler : None or selene_sdk.samplers.file_samplers.FileSampler, optional
+    train_sampler : selene_sdk.samplers.file_samplers.FileSampler or
+            selene_sdk.samplers.dataloader.DataLoader
+        Load your training data as a `FileSampler` or `DataLoader`
+    validate_sampler : FileSampler or DataLoader
+        The validation dataset file sampler or data loader.
+    targets : list(str)
+        The list of targets the model should predict
+    test_sampler : None or FileSampler or DataLoader, optional
         Default is None. The test file sampler is optional.
     mode : str, optional
         Default is "train". Must be one of `{train, validate, test}`. The
         starting mode in which to run the sampler.
     save_datasets : list(str), optional
```

Comment:
maybe we should just remove this from Sampler if it doesn't seem useful here? I think it's not always useful for other samplers either right?

Reply:
or just allow save datasets for online samplers only??

Reply:
is there any randomization to the sampled datasets that you think people would find useful having saved? we could just save a file of indices for each mode but idk if that's hard to track especially with dataloader